core/src/main/java/jenkins/model/Nodes.java

```diff
@@ -256,6 +259,14 @@ public void run() {
                     if (c != null) {
                         c.recordTermination();
                         c.disconnect(OfflineCause.create(hudson.model.Messages._Hudson_NodeBeingRemoved()));
+                        for(Executor executor: c.getAllExecutors()){
+                            executor.interrupt(Result.ABORTED, new CauseOfInterruption() {
+                                @Override
+                                public String getShortDescription() {
+                                    return "Node was removed by " + User.current().getId();
```

Comment:
```suggestion
                                    return "Node was removed by " + (User.current() == null ? "null" : User.current().getId());
```

Reply:
Done through the command line. GitHub was not letting me commit this through the UI. :man_shrugging:. Thanks Matt!